cmd/crank/beta/top/top.go

```diff
@@ -0,0 +1,252 @@
+		if err != nil {
+			return errors.Wrap(err, errAddingPodMetrics)
+		}
+		var totalCPUUsage, totalMemoryUsage int64
+		for _, container := range podMetrics.Containers {
+			totalCPUUsage += container.Usage.Cpu().ScaledValue(resource.Milli)
+			totalMemoryUsage += container.Usage.Memory().ScaledValue(resource.Mega)
+		}
+
+		crossplanePods[i].CPUUsage = totalCPUUsage
+		crossplanePods[i].MemoryUsage = fmt.Sprintf("%dMi", totalMemoryUsage)
+	}
+	logger.Debug("Metrics added to Crossplane pods")
+
+	if err != nil {
+		return errors.Wrap(err, errGetPodMetrics)
+	}
+	logger.Debug("Added metrics to Crossplane pods")
+	if c.Split {
+		printPodsByCategory(crossplanePods)
+	} else {
+		headers := defaultPrinterRow{
+			podType:   "TYPE",
+			namespace: "NAMESPACE",
+			name:      "NAME",
+			cpu:       "CPU(cores)",
+			memory:    "MEMORY(bytes)",
+		}
+		if _, err := fmt.Fprintln(tw, headers.String()); err != nil {
+			return errors.Wrap(err, errWriteHeader)
+		}
+
+		for _, pod := range crossplanePods {
+			row := defaultPrinterRow{
+				podType:   pod.PodType,
+				namespace: pod.PodNamespace,
+				name:      pod.PodName,
+				cpu:       fmt.Sprintf("%d", pod.CPUUsage),
+				memory:    pod.MemoryUsage,
+			}
+			if _, err := fmt.Fprintln(tw, row.String()); err != nil {
+				return errors.Wrap(err, errWriteRow)
+			}
+		}
+	}
+
+	if err := tw.Flush(); err != nil {
+		return errors.Wrap(err, errFlushTabWriter)
+	}
+	return nil
+}
+func printPodsByCategory(pods []topMetrics) {
+	categorySummaries := make(map[string]struct {
+		Count       int
+		TotalCPU    int64
+		TotalMemory string
+	})
+
+	for _, pod := range pods {
+		summary := categorySummaries[pod.PodType]
+		summary.Count++
+		summary.TotalCPU += pod.CPUUsage
+
+		// Inline memory aggregation logic
+		if summary.TotalMemory == "" {
+			summary.TotalMemory = pod.MemoryUsage
+		} else {
+			// Concatenate for simplicity; replace with actual aggregation logic
+			summary.TotalMemory += "+" + pod.MemoryUsage
+		}
+
+		categorySummaries[pod.PodType] = summary
+	}
+
+	for category := range categorySummaries {
+		fmt.Printf("\n== %s Pods: ==\n\n", capitalizeFirst(category))
+		for _, pod := range pods {
+			if pod.PodType == category {
+				fmt.Printf("namespace: %s\nname: %s\ncpu: %d\nmemory: %s\n\n",
+					pod.PodNamespace, pod.PodName, pod.CPUUsage, pod.MemoryUsage)
+			}
+		}
+	}
+
+}
+func capitalizeFirst(s string) string {
+	if s == "" {
+		return ""
+	}
+	return strings.ToUpper(s[:1]) + s[1:]
+}
+func getCrossplanePods(pods []v1.Pod) []topMetrics {
+	metricsList := make([]topMetrics, 0)
+	for _, pod := range pods {
+		labels := pod.GetLabels()
+
+		var podType string
+		isCrossplanePod := false
+		for labelKey, labelValue := range labels {
+			switch {
+			case strings.HasPrefix(labelKey, "pkg.crossplane.io/provider"):
+				podType = "provider"
+				isCrossplanePod = true
+			case strings.HasPrefix(labelKey, "pkg.crossplane.io/function"):
+				podType = "function"
+				isCrossplanePod = true
```

Comment:
It would be neat if we didn't have to update this command to support new package types in future. Would it be possible to derive the pod type from the last segment of the `pkg.crossplane.io/...` annotation?

Reply:
Good idea! Do you know if all the other crossplane pods with automatically get the `pkg.crossplane.io/...` label? For example, the _core_ crossplane pods don't have this label, that's why I used `app.kubernetes.io/part-of`

```bash
crossplane-75575fcf5d-94wlh                                  1/1     Running   0             70m   app.kubernetes.io/component=cloud-infrastructure-controller,app.kubernetes.io/instance=uxp,app.kubernetes.io/managed-by=Helm,app.kubernetes.io/name=crossplane,app.kubernetes.io/part-of=crossplane,app.kubernetes.io/version=1.14.5-up.1,app=crossplane,helm.sh/chart=universal-crossplane-1.14.5-up.1,pod-template-hash=75575fcf5d,release=uxp
crossplane-rbac-manager-77f845f758-rvngs                     1/1     Running   0             70m   app.kubernetes.io/component=cloud-infrastructure-controller,app.kubernetes.io/instance=uxp,app.kubernetes.io/managed-by=Helm,app.kubernetes.io/name=crossplane,app.kubernetes.io/part-of=crossplane,app.kubernetes.io/version=1.14.5-up.1,app=crossplane-rbac-manager,helm.sh/chart=universal-crossplane-1.14.5-up.1,pod-template-hash=77f845f758,release=uxp
```

Reply:
Only pods of package runtimes (e.g. today, functions and providers) will get that label. So it won't cover all cases but it would at least be nice to cover the package-pod case generally.

Reply:
It was easier to implement than I thought, done :heavy_check_mark: